### DISCUSSION TITLE
Automatically disable node scheduling and request eviction on node cordon

### DISCUSSION THREAD

**Discussion by paravibe** [longhorn/longhorn#5736] [Q&A] [answered]
Is there a way to automatically flag node and it's disks as non schedulable and request eviction once cluster node is cordoned?
This is useful during Kubernetes version upgrade or any other maintenance work. Right now I need to do it manually every time.

**Comment by c3y1huang** [comment]
Doesn't node drain work for your case?

**Comment by paravibe** [comment]
No, unfortunately it does nothing

**Reply by c3y1huang** [reply]
It will be helpful if we can get more context around the scenario.

Can you clarify what you're attempting to evict and at which steps you need to perform manually while referring to the [node maintenance guide](https://longhorn.io/docs/1.4.0/volumes-and-nodes/maintenance/)?

**Comment by paravibe** [comment]
Ok, so recently I've upgraded kube version from 1.25 to 1.26 and during this process nodes in the pool are marked as cordoned and Longhorn immediately marks nodes as non schedulable BUT doesn't evict all the replicas from the nodes that are about to be deleted so I need to do it manually every time. If I don't do this I simply lose all data that was stored on those nodes.

**Reply by innobead** [reply, marked as answer]
I guess your upgrade is not in-place, so the drained node will be deleted, right? If yes, what you are looking for is https://github.com/longhorn/longhorn/issues/2238.

**Reply by innobead** [reply]
cc @PhanLe1010

**Reply by paravibe** [reply]
Yes, drained node will be deleted. Thanks for pointing to the request.
